pdr_backend/subgraph/subgraph_payout.py

```diff
@@ -0,0 +1,143 @@
+from typing import List
+from enforce_typing import enforce_types
+
+from pdr_backend.subgraph.core_subgraph import query_subgraph
+from pdr_backend.util.networkutil import get_subgraph_url
+from pdr_backend.subgraph.payout import Payout
+
+
+@enforce_types
+def get_payout_query(
+    asset_ids: List[str], start_ts: int, end_ts: int, first: int, skip: int
+) -> str:
+    """
+    Constructs a GraphQL query string to fetch prediction slot data for
+    specified assets within a slot range.
+
+    Args:
+        prediction_ids: A list of prediction identifiers to include in the query.
+        first: The number of records to fetch per query (pagination limit).
+        skip: The number of records to skip (pagination offset).
+
+    Returns:
+        A string representing the GraphQL query.
+    """
+
+    # asset_ids_str = str(asset_ids).replace("[", "[").replace("]", "]").replace("'", '"')
+    where_query_arr = []
+
+    for asset_id in asset_ids:
+        where_query_arr.append(
+            """
+                {
+                    timestamp_gte: %s,
+                    timestamp_lte: %s,
+                    prediction_contains: "%s"
+                }
+            """
+            % (start_ts, end_ts, asset_id)
+        )
```

Comment:
I'm not sure why this has to be looped...

I.e. we should be able to use a single where clause
![image](https://github.com/oceanprotocol/pdr-backend/assets/69865342/d894bba9-629f-447a-9ac7-7b4571cbafc6)

That generates the following query for multiple assets
```
query {
            predictPayouts (
            first: 10
            skip: 1
            where: {
        {
            timestamp_gte: 1622547000,
            timestamp_lte: 1622548800,
            prediction_contains: ['0x18f54cc21b7a2fdd011bea06bba7801b280e3151', '0x33334cc21b7a2fdd011bea06bba7801b280e3151']
        }
    }
        ) {
                id
                timestamp
                payout
                prediction {
                    user {
                        id
                    }
                    slot {
                        id
                        predictContract{
                            id
                            token{
                                name
                            }
                        }
                    }
                }
            }
        }
```

as defined in this test example

![image](https://github.com/oceanprotocol/pdr-backend/assets/69865342/159a4711-37b2-4acf-b77c-50b75fafa741)

Reply:
I took another look at your comment, @idiom-bytes. The "aaset_ids" argument is a list of strings that represent the IDs of the contracts we want to retrieve. Unfortunately, the "predictPayouts" query's "where" clause does not have a contract filter. We could use a nested query, like the one below, but our subgraph system does not allow it:

```
where: {
    prediction_: {
        slot_: {
            predictContract_in: ["0xfeed1", "0xfeed2"]
        }
    }
}
```

However, the "prediction_contains" argument searches for text within prediction IDs. It only allows strings. The structure of a prediction ID is as follows:

```
{contract address}-{slot}-{user}
```

Therefore, if the contract address is present in the prediction ID, we can retrieve it.

Reply:
Looked into this and I agree with @kdetry, we can't just pass the list of addresses with the way subgraph it's structured right now